noir-projects/aztec-nr/aztec/src/context/inputs/private_context_inputs.nr

```diff
@@ -10,5 +10,6 @@ struct PrivateContextInputs {
     call_context : CallContext,
     historical_header: Header,
     private_global_variables: PrivateGlobalVariables,
+    start_side_effect_counter: u32,
```

Comment:
@LeilaWang came with the suggestion to remove it from `CallContext` entirely, which I've attempted to do in #5177 

> Why did we need to add this here instead of continuing to use the one in CallContext?

The value in `call_context` is either the `start_side_effect_counter` for private calls and public calls in public execution, while it is the `enqueued_side_effect_counter` for enqueued public calls.

The real `start_side_effect_counter` is the one returned by `context.finish()` and for enqueued calls it will be different than what's in `call_context` (I enqueue it at 5 but run the function at 20). Because of the hashing that goes on in the kernel I can't modify `call_context.side_effect_counter` for enqueued calls.

In #5177 we remove `call_context.side_effect_counter` and rely on public inputs.